### ISSUE TITLE
Improve reports

### ISSUE THREAD

**Issue by lonix1** [invoiceninja/admin-portal#259]
I have a blocking issue due to invoiceninja/invoiceninja/issues/5484 which is apparently fixed in 49 but not for me. The only thing I can think of is that during one of the various migrations my database was corrupted. So my installation is unusable and unsure how to proceed.

An idea is to export all data, then import. But the export feature is not yet ready.

A workaround is to export from the reporting tool, manually clean the csv data, then import via the import tool. However the reporting tool doesn't export everything. It seems easier to fix the reporting tool than add a new export tool.

This would help others as well - I've found other issues regarding import/export, so it would be a useful workaround for people until a proper export feature is added.

I'm documenting what I've found so far:

### products report
missing fields:
- notes

incorrectly formatted fields:
- quantity should not have leading currency, e.g. "$1" -> "1"

### clients report
missing fields:
- first_name
- last_name
 
incorrectly named fields (may lead to errors during export/import):
- "Custom Value" -> "Custom Value 1"
- "Custom Value" -> "Custom Value 2"

### quote line items report
doesn't exist (only *invoice* line items are exportable)
because of this quotations cannot be exported/imported

### quotes report
haven't got that far yet

### invoice line items report
haven't got that far yet

### invoices report
haven't got that far yet

### payments report
haven't got that far yet

### other reports
I don't use other features so haven't (/won't) checked anything else

### general export rules
As per typical csv, some general guidelines (think of how excel works):
- numeric fields should not have leading currency
- numeric fields should not have decimal separator
- numeric fields should only have trailing decimals if necessary: "1.00" -> "1", but "1.50" is ok

Is any of this possible?

**Comment by hillelcoren** [comment]
Thanks for the feedback! I'm happy to try to work through any issues you find, please add comments though rather than update.

**Comment by lonix1** [comment]
Ok... I'll work through the other reports later and add comments below.

Also I'll post my excel file with formulas to help others "massage" their data once its exported. Some things are easier to handle in excel rather than in the app.

**Comment by lonix1** [comment]
The underlying problem was fixed in v53 (invoiceninja/invoiceninja/issues/5484), so this is not critical to me as it was for the last few days.

But I still think it's a good idea to look into this until there's a better way (i.e. standalone export feature). I see you already made some changes, thanks I think they help (especially the new quote line items report).

I also realized there's another temporary solution - some javascript that pulls data from the backend API. I started working on that before the latest fix was released. The benefit there is the returned json can be easily formatted to fit the needs of the import tool.

(I don't know if you want to close this for now? I'll update it anyway when I get round to working on the reports again as discussed above.)
